packages/react/src/hooks/useOutsideClick.ts

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from "react";
+
+function useOutsideClick<T extends HTMLElement>(
+  ref: React.RefObject<T>,
```

Comment:
You could combine `ref` and `exceptions` into a single parameter called `elements` or `inside` for example because you are detecting for clicks outside of both.